src/poliastro/contrib/extract_czml.py

```diff
@@ -0,0 +1,288 @@
+import json
+
+import numpy as np
+from astropy import units as u
+from astropy.coordinates import CartesianRepresentation
+from astropy.time import Time
+
+from poliastro.contrib.czml_extract_default_params import DEFAULTS
+
+
+class ExtractorCZML:
+    """A class for extracting orbitary data to Cesium"""
+
+    def __init__(self, orbit, start_epoch, end_epoch, N, fixed_rf=True):
+        """
+        Orbital constructor
+
+        Parameters
+        ----------
+        orbit: poliastro.Orbit
+            Orbit to be extracted
+        start_epoch: ~astropy.time.core.Time
+            Starting epoch
+        end_epoch: ~astropy.time.core.Time
+            Ending epoch
+        N: int
+            Number of sample points
+        fixed_rf: bool
+            Determines whether the reference frame is fixed
+            When set to false, it assumes the frame moving
+            with constant velocity.
+        """
+        self.czml = dict()
+        self.orbits = {1: [orbit, N, orbit.period, orbit.epoch]}
+
+        self.i = 1  # Current index id, used for insertion of new elements
+
+        self.fixed_rf = fixed_rf
+        self.start_epoch = ExtractorCZML.format_date(start_epoch.value)
+        self.end_epoch = ExtractorCZML.format_date(end_epoch.value)
+        self.init_czml()
+
+    def parse_dict_tuples(self, path, tups):
+        """
+        Parameters
+        ----------
+        path : list (val)
+            Dictionary path to insert to
+        tups : list (val, val)
+            Tuples to be assigned
+        """
+
+        # We only want to pass a reference czml, then by modifying our reference, we'll be modifying our base dictionary
+        # which allows us to walk through a path of arbitrary length
+
+        curr = self.czml
+        for p in path:
+            curr = curr.setdefault(p, {})
+        for t in tups:
+            curr[t[0]] = t[1]
+
+    def init_orbit_packet(self, i):
+        """
+        Sets the default values for a given orbit
+
+        Parameters
+        ----------
+        i : int
+            Index of referenced orbit
+        """
+        for t_key, t_val in DEFAULTS:
```

Comment:
Instead of storing the `DEFAULTS` as a list of tuples externally and then using `parse_dict_tuples` on it, wouldn't it be simpler to have some default `._czml` and then add properties to it?

Reply:
That's a really great idea and maybe far more intuitive. I initially had the idea of having some sort of choice between ``DEFAULTS`` but it didn't pan out.